### PR TITLE
ui: Move /status & /silences to API v2

### PR DESCRIPTION
**Pull request by mxinden**
This patch makes the Alertmanager UI (/status & /silences) use the
api/v2 endpoint. In addition it adds logic to generate the elm side data
model based on the OpenAPI specification.

In addition requesting alerts from api v1 is done in a separate commit to be
able to revert it once alerts also come from api v2.

### CODE REVIEW

File: ui/app/src/Views/SilenceList/Views.elm [hunk 1]
```diff
@@ -97,6 +100,23 @@ states =
...
+
+
+filterSilenceByState : State -> Silence -> Bool
+filterSilenceByState state silence =
+    -- TODO: Can this be done cleaner?
+    Maybe.withDefault False <| Maybe.map (.state >> (==) state) <| .status silence
+
+
+
+{-
```
Comment (stuartnelson3) [inline ui/app/src/Views/SilenceList/Views.elm]: This can be removed?

**Comment by stuartnelson3** [comment]
Thanks for doing all this. We haven't touched the elm code in a while, but it's really nice coming back to this language and feeling confident in the changes we make thanks to the type system :)

File: ui/app/src/Data/Alert.elm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import Data.Receiver as Receiver exposing (Receiver)
+import DateTime exposing (DateTime)
+import Dict exposing (Dict)
+import Json.Decode as Decode exposing (Decoder)
+import Json.Decode.Pipeline exposing (optional, required)
+import Json.Encode as Encode
+
+
+type alias Alert =
+    { startsAt : Maybe DateTime
```
Comment (mxinden) [inline ui/app/src/Data/Alert.elm]: These will change once I move `/alerts` to the new API as well.

**Comment by stuartnelson3** [comment]
Looks good to me!

**Comment by mxinden** [comment]
Thanks @w0rm and @stuartnelson3. I will follow up with moving the `/alerts` page as well to API v2.

**Pull request merged**